src/OptimisticSessionHandler.php

```diff
@@ -83,22 +88,36 @@ public function writeIfSessionChanged()
 
         if($needWrite) {
             $this->lock = true;
-            ob_start();
-            session_start();
-            ob_clean();
+            //We need @session_start() because we can't send session cookie more then once.
+            @session_start();
             $sameOldAndNew = $this->array_compare_recursive($_SESSION, $oldSession);
             $sameOldAndNew = $sameOldAndNew || $this->array_compare_recursive($oldSession, $_SESSION);
 
             if($sameOldAndNew) {
                 $_SESSION = $currentSession;
             } else {
-                $tab1 = array_merge($_SESSION, $currentSession);
-                $tab2 = array_merge($currentSession, $_SESSION);
-                if(!$this->array_compare_recursive($tab1, $tab2)) {
-                    throw new \Exception('Conflicts in sessions changes');
-                }
+                $keys = array_keys(array_merge($_SESSION, $currentSession, $oldSession));
 
-                $_SESSION = $tab1;
+                foreach ($keys as $key) {
+                    $base = isset($oldSession[$key])?:null;
+                    $mine = isset($currentSession[$key])?:null;
+                    $theirs = isset($_SESSION[$key])?:null;
+                    if ($base != $mine && $base != $theirs && $mine != $theirs) {
+                        foreach($this->conflictRules as $regex => $conflictRule){
+                            if(preg_match($regex, $key)){
+                                if ($conflictRule == 1) {
+                                    $_SESSION[$key] = $mine;
+                                } elseif ($conflictRule == -1) {
```

Comment:
Instead of -1, 0, and +1, you should create constants:
self::IGNORE
self::OVERWRITE
self::FAIL (default)